### PR TITLE
SW-4625 Make annual reports start December 1

### PR DESCRIPTION
**Pull request by nickgraz**
- Make the 'isAnnual' check look a the quarter two months before the current date, allowing annual reports to be created on December 1

### CODE REVIEW

**Comment by nickgraz** [comment]
Current dependencies on/for this PR:
* `main`
  * **PR #1584** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/1584?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-server/1584?utm_source=stack-comment).

**Comment by nickgraz** [comment]
@sgrimm this is my first crack at this. I think I need to do something in [here](https://github.com/terraware/terraware-server/blob/3aa26dfb8023770bee2e005a455c49a277425cff/src/main/kotlin/com/terraformation/backend/report/db/ReportStore.kt#L394) but I am unsure. Or perhaps I just need to update the `getLastQuarter` function [here](https://github.com/terraware/terraware-server/blob/3aa26dfb8023770bee2e005a455c49a277425cff/src/main/kotlin/com/terraformation/backend/report/db/ReportStore.kt#L627) to also only look 2 months behind?

**Comment by sgrimm** [comment]
I had a crack at this too and took a different approach: #1583

File: src/main/kotlin/com/terraformation/backend/report/ReportService.kt [hunk 1]
```diff
@@ -214,7 +214,7 @@ class ReportService(
       body: LatestReportBodyModel? = null
   ): LatestReportBodyModel {
     val organization = organizationStore.fetchOneById(organizationId)
-    val isAnnual = body?.isAnnual ?: (ZonedDateTime.now(clock).minusMonths(3).quarter == 4)
+    val isAnnual = body?.isAnnual ?: (ZonedDateTime.now(clock).minusMonths(2).quarter == 4)
```
Comment (sgrimm) [inline src/main/kotlin/com/terraformation/backend/report/ReportService.kt:217]: I think this won't be quite right if this code is creating a report in March (say, for an org whose reporting was enabled late).

**Comment by nickgraz** [comment]
Closing this out for https://app.graphite.dev/github/pr/terraware/terraware-server/1583/SW-4625-Create-Q4-report-on-December-1?utm_source=gt-slack-notif

**Pull request closed without merging**